### PR TITLE
Multifitter implementation

### PR DESCRIPTION
**Pull request by wardsimon**
This is a basic multi-fitter implementation based on the work of @arm61 and the 'xarray' dataset multi-fitting interface.

This pull request allows for the optimisation of ND datasets **AND** multiple ND datasets. The api remains unchanged (though it should be tested on downstream implementations). The only difference is that when using the `MultiFitter` class, an array of `FitResults` are returned.

### CODE REVIEW

**Comment by wardsimon** [comment]
Notes on what needs to change:
- goodness of fit is currently set to reduced-chi2 not chi2 (that is the job of the reduced_chi2 field)
- computation of Chi2 needs to be reimplemented for multi-fit
- Weights need to have reshaping applied.

File: easyCore/Utils/typing.py [hunk 1]
```diff
@@ -2,10 +2,17 @@
...
+
+# Template Constraint
 from easyCore.Fitting.Constraints import C
+
+# Variable such as Descriptor, Parameter
 from easyCore.Objects.Variable import V
-from easyCore.Objects.ObjectClasses import B, BV
\ No newline at end of file
+
+# Base object from BaseObj (B) a BaseObject OR Varialbe (BV)
```
Comment (arm61) [inline easyCore/Utils/typing.py:17]: Typo
```suggestion
# Base object from BaseObj (B) a BaseObject OR Variable (BV)
```

**Review by arm61 — APPROVED** [review]
Looks good (the parts that I understand). Only comment would be that there are some empty docstring args, but I imagine you are getting to that. 

Do you want me to test on EasyReflectometry (would be a job for tomorrow)?

**Comment by wardsimon** [comment]
With b60be50 docstrings and GOF is now up to date

**Comment by arm61** [comment]
I will test on EasyReflectometry tomorrow morning!

**Pull request merged**